src/genie/core/record/alignment-box.cc

```diff
@@ -30,15 +30,16 @@ void AlignmentBox::write(util::BitWriter& writer) const {
 // ---------------------------------------------------------------------------------------------------------------------
 
 AlignmentBox::AlignmentBox(ClassType type, uint8_t as_depth, uint8_t number_of_template_segments,
-                           util::BitReader& reader) {
-    mapping_pos = reader.read<uint64_t>(40);
+                           util::BitReader& reader)
+    : splitAlignmentInfo(type == ClassType::CLASS_HM ? 0 : number_of_template_segments - 1) {
+    mapping_pos = reader.readBypassBE<uint64_t, 5>();
     alignment = Alignment(as_depth, reader);
 
     if (type == ClassType::CLASS_HM) {
         return;
     }
     for (size_t tSeg = 1; tSeg < number_of_template_segments; tSeg++) {
-        splitAlignmentInfo.push_back(AlignmentSplit::factory(as_depth, reader));
+        splitAlignmentInfo[tSeg - 1] = AlignmentSplit::factory(as_depth, reader);
```

Comment:
Why FIFO with offset 1?

Reply:
*Why add to front and not to the back?

Reply:
*Sorry, i see what you did there.
You reserve the memory first then add the elements

Reply:
Only the second alignment is a split alignment, the first alignment is already included in the "Alignment Box". However number_of_template_segments includes both of them. See the structure of MPEG-G record in part 2